language-design.md

```diff
@@ -17,7 +17,78 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
-
+> I have said in the past, and will say now, that I think it would be a good thing
+> for the Java programming language to ... let the user define overloaded operators.
+> Just as a user can code methods that can be used in just the same way as
+> methods that are built in, the user ought to have a way to define operators for user
+> defined classes that can be used in just the same way as operators that are built in. [Steele, 1998]
+
+I never really considered the importance of operator overloading before, but Steele makes a very convincing argument for them:
+opertor overloading allows an extension of a language (e.g., a class or package) to be as concise, familiar, and 'primitive'
+as the underlying language. For example, in languages such as Java where operator overloading is not supported,
+users must write an `add` function to represent the addition of two objects; the resulting code looks something like
+`add(obj1, obj2)`, which gets the job done but isn't as simple and pleasing as the operator overloaded alternative `obj1 + obj2`.
+C++ and Python are a couple languages that _do_ support operator overloading; and realizing that Python--
+a language that I think people tend to not take very seriously for professional development--
+supports a particularly preactical feature that Java does not, made me have a lot more respect for Python.
+I had always thought of Python as more of a language for learners because (I hear) it has so much overhead
```

Comment:
Overhead is relevant, sort of.  The main complaint is that Python is "too slow".  The first issue with this is rather pedantic - Python is a language.  A language can't be slow.  and _implementation_ can be, and usually people are referring to the (standard) CPython implementation.  However, there are other implementations (Jython, IronPython, and PyPy for example) that are much faster. 

The second issue is that "too slow" is a **horrible metric**.  The real question you should be asking is "Is my code fast enough?"  If so, then you're done.  If not, then you need to improve it.  Python makes this very easy by providing a lot of ways to hook in C/C++ code (for the parts that need micro-optimizations) to your existing Python code, or using Cython (a DSL that combines the two) to have readable, Python code that compiles to C.